packages/ai/src/generate-text/generate-text.ts

```diff
@@ -600,16 +600,22 @@ A function that attempts to repair a tool call that failed to parse.
 
         const lastStep = steps[steps.length - 1];
 
-        return new DefaultGenerateTextResult({
-          steps,
-          resolvedOutput: await output?.parseOutput(
+        // parse output if the last step was finished with "stop":
+        let resolvedOutput;
+        if (lastStep.finishReason === 'stop') {
```

Comment:
@lgrammel fyi update it to stop as discussed, doesn't break anything as far as I can tell.